src/main/java/com/github/manolo8/darkbot/core/objects/facades/AssemblyMediator.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    @Override
+    public boolean isFilterDropDownOpen() {
+        return isFilterDropDownOpen;
+    }
+
+    @Override
+    public List<? extends AssemblyAPI.Recipe> getRecipes() {
+        return recipes;
+    }
+
+    @Override
+    public List<? extends AssemblyAPI.RowFilter> getRowFilters() {
+        return rowSettings;
+    }
+
+    @Override
+    public List<? extends AssemblyAPI.Filter> getFilters() {
+        return filters;
+    }
+
+    public static class Recipe extends Auto implements AssemblyAPI.Recipe {
+        public String recipeId = "";
+        public List<String> rewards = new ArrayList<>();
+        public final ObjArray rewardsArr = ObjArray.ofVector(true);
+
+        public List<ResourceRequired> resourcesRequired = new ArrayList<>();
+        private final ObjArray resourcesRequiredArr = ObjArray.ofVector(true);
+
+        public boolean isCraftable = false;
+
+        @Override
+        public void update() {
+            if (address == 0) return;
+            isCraftable = API.readBoolean(address + 0x20);
+
+            long itemVoAddress = Main.API.readMemoryPtr(address + 0x58);
+            recipeId = API.readMemoryString(itemVoAddress, 0x48);
+
+            long rewardsArrAddress = Main.API.readMemoryPtr(address + 0x60);
+            rewardsArr.update(rewardsArrAddress);
+            rewards.clear();
+            rewardsArr.forEach(ptr -> rewards.add(API.readMemoryString(ptr, 0x48)));
+
+            long resourcesRequiredArrAddress = Main.API.readMemoryPtr(address + 0x50);
+            resourcesRequiredArr.update(resourcesRequiredArrAddress);
+            resourcesRequiredArr.sync(resourcesRequired, ResourceRequired::new);
+        }
+
+        @Override
+        public String getRecipeId() {
+            return recipeId;
+        }
+
+        @Override
+        public List<? extends String> getRewards() {
+            return rewards;
+        }
+
+        @Override
+        public List<? extends AssemblyAPI.ResourceRequired> getResourcesRequired() {
+            return resourcesRequired;
+        }
+
+        @Override
+        public boolean isCraftable() {
+            return isCraftable;
+        }
+
+        @Override
+        public String toString() {
+            return Recipe.class.getSimpleName() + " - " + recipeId + " - " + isCraftable;
+        }
+    }
+
+
+    public static class ResourceRequired extends Auto implements AssemblyAPI.ResourceRequired {
+        public String resourceId = "";
+        public double amountRequired, amountRequiredBackup = -1.0;
+
+        public void update() {
+            if (address <= 0) return;
+            long itemVoAddress = Main.API.readMemoryPtr(address + 0x28);
+            resourceId = API.readMemoryString(itemVoAddress, 0x48);
+            amountRequired = API.readDouble(address + 0x30);
+            //this also gives back same value, not sure which is correct
+            amountRequiredBackup = API.readDouble(address + 0x38);
```

Comment:
Is it possible that one is the unitary amount required, while the other is the total amount required? so if a recipe needs 10 material, but you're trying to craft 2 of the item, one will be 10, while the other is 20. If so, can we get that reflected?

Reply:
Checked to validate that ingame but doesnt reflect to be the case